### DISCUSSION TITLE
Changing the zx's shell in Windows

### DISCUSSION THREAD

**Discussion by Mojtaba-NA** [johnlindquist/kit#1130] [Q&A] [answered]
Hi, is there any way to change the shell that zx uses in windows?
I've tried $`ls --shell=path-to-bash` and putting $.shell = 'path-to-bash' before them but neither works

**Comment by johnlindquist** [comment]
@Mojtaba-NikAeen This is all I could find:

https://github.com/google/zx/issues/523#issuecomment-1270722227

Just to be sure, in your `~/.kit/package.json`, is `zx` on `7.1.1`?

If you need to upgrade that, use:

(I'm assuming you're using windows bash):
```bash
cd ~/.kit && PATH=~/.knode/bin:$PATH npm i zx
```

**Reply by Mojtaba-NA** [reply]
I've upgraded to 7.1.1 and now it uses powershell instead of cmd and still won't let me change it to anything else.

**Reply by johnlindquist** [reply]
@Mojtaba-NikAeen You can try the latest pre-release of Script Kit. It probably won't fix it, but 🤷‍♂️

https://github.com/johnlindquist/kitapp/releases/tag/v1.49.4

**Reply by Mojtaba-NA** [reply]
I'm not in a hurry and I'm pretty sure this is zx's issue (I'll tinker with that) so I'll just wait for the next release.
Thank you for your help. 
Love your app, It's amazing.

**Comment by Mojtaba-NA** [comment, marked as answer]
I've installed zx in .kenv folder and imported $ directly from zx and everything seems to be working fine.
small thing I've noticed is that `$.shell` returns undefined in script kit but returns the path that it detects automatically when running separately using `zx ./script.mjs`.

**Reply by fcrespo82** [reply]
@Mojtaba-NikAeen Can you show me an example on how you achieved that?
I'm trying what you said but my script hangs and never finishes.

**Reply by Mojtaba-NA** [reply]
@fcrespo82
When you edit a script, it opens up a folder called `.kenv`, in that folder run the command `npm i zx`.
then:
`import { $ } from 'zx'`

now use that $ to run commands instead of the global $ that scriptkit provides.
```await $`ls -l` ```

to change the shell,  put this at the top of your script (or just a line above where you're actually using the $):
`$.shell = 'path-to-executable'`

**Reply by fcrespo82** [reply]
Thanks.

My mistake was that even when I was installing zx on .kenv, I was using as `const zx = await npm('zx')`
